### ISSUE TITLE
[BUG] Automated Weapon attacks with ammunition don't work after updating to dnd5e 2.4.1

### ISSUE THREAD

**Issue by TheAlexPlus** [roth-michael/FoundryVTT-RestRecovery#184]
**Describe the bug**
Rest Recovery doesn't seem to be playing nice with MidiQOL after updating to dnd5e 2.4.1  (Running Foundry v10).   Any attack roll with an associated ammunition fails to roll.
Each module works on their own, but since MidiQOL is supposed to involve attacks, and Rest Recovery seems unrelated, I figured the issue was on the Rest Recovery side.

**To Reproduce**
1. Have MidiQOL and Rest Recovery installed.
2. Turn on full automation.. or any level of automation that fast forwards the attack rolls.
3. Give player a ranged weapon that uses ammo and link the ammo to the weapon.
4. Target enemy and attempt to attack
5. See that nothing happens.

**Expected behavior**
I expected it to attack and continue through the roll automation.

**Setup:**
 - Foundry Version: 10.312
 - System Version: dnd5e 2.4.1
 - Rest Recovery Version: 1.4.17
 - Browser & version: Version 120.0.6099.130 (Official Build) (64-bit)

**Active modules:**
 - Rest Recovery
 - libwrapper
 - Midi-QOL
 - socketlib

**Additional context**
The error that comes up in the console:
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'push')
    at Item5e._handleConsumeResource (item.mjs:1144:25)
    at Item5e._getUsageUpdates (libwrapper.js:310:33)
    at 🎁call_wrapper [as call_wrapper] (libWrapper-wrapper.js:591:14)
    at 🎁CONFIG.Item.documentClass.prototype._getUsageUpdates#0 (libWrapper-wrapper.js:189:20)
    at Item5e.rollAttack (item.mjs:1323:26)
    at 🎁call_wrapped [as call_wrapped] (libWrapper-wrapper.js:507:22)
    at Item5e.doAttackRoll (itemhandling.js:470:21)
    at async Workflow._next (workflow.js:405:6)
    at async Workflow.next (workflow.js:248:10)
    at async Workflow.next (workflow.js:248:10)
    at async Workflow.next (workflow.js:248:10)
    at async Workflow.next (workflow.js:248:10)
    at async Workflow.next (workflow.js:248:10)

**Comment by Haxxer** [comment]
Perhaps... update Rest Recovery?

**Comment by TheAlexPlus** [comment]
I did.  that's the first thing I do when I update the system.  But it looks like the v10 version of Rest Recovery hasn't been updated for dnd5e 2.4.1.  Rest Recovery 1.4.17 is the farthest i can update within v10

**Comment by roth-michael** [comment]
Fixed once on 3.1.x. I won't be making any pre-3.1 releases (especially as an incompatibility with midi would mean working with outdated midi as well), so will be marking this closed (and fixed - as technically this isn't a bug with all the most recent updates). I hope whatever's keeping you on V10 lets you move forward soon!

**Comment by TheAlexPlus** [comment]
I finally pulled the trigger on moving forward to v11 yesterday, but being forced to use dnd 3.0 is the problem now. Not everything works with it.
I understand the headache with having to support so many versions of stuff, it just sucks when your core modules don't work with each other.  I just started a Tomb of Annihilation campaign which is pretty resource intensive... so its tough to have to choose between automated rests/resource consumption and having your ammunition accurately subtract.

I'm just sort of venting.  I appreciate all the work you do!
